implementations/micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalFxNamingConvention.java

```diff
@@ -80,22 +72,47 @@ public String name(String name, Meter.Type type, @Nullable String baseUnit) {
     @Override
     public String tagKey(String key) {
         String conventionKey = delegate.tagKey(key);
+        conventionKey = PATTERN_TAG_KEY_DENYLISTED_CHARS.matcher(conventionKey).replaceAll("_");
+
+        if (conventionKey.length() < 1) {
+            return conventionKey;
+        }
 
-        conventionKey = START_UNDERSCORE_PATTERN.matcher(conventionKey).replaceAll(""); // 2
-        conventionKey = SF_PATTERN.matcher(conventionKey).replaceAll(""); // 2
+        int i = 0;
+        while (conventionKey.length() > i) {
+            if (conventionKey.startsWith("sf_", i)) {
+                i += 3;
+                continue;
+            }
+            if (conventionKey.startsWith("_", i)) {
+                i += 1;
+                continue;
+            }
+            break;
+        }
 
-        conventionKey = PATTERN_TAG_KEY_DENYLISTED_CHARS.matcher(conventionKey).replaceAll("_");
-        if (!START_LETTERS_PATTERN.matcher(conventionKey).matches()) { // 3
+        if (i > 0) {
+            conventionKey = conventionKey.substring(i);
+            if (conventionKey.length() < 1) {
+                return conventionKey;
+            }
+        }
+
+        if (!Character.isLetter(conventionKey.charAt(0))) {
+            logger.log(conventionKey
+                    + " doesn't adhere to SignalFx naming standards. Prefixing the tag/dimension key with 'a'.");
             conventionKey = "a" + conventionKey;
         }
-        if (PATTERN_TAG_KEY_DENYLISTED_PREFIX.matcher(conventionKey).matches()) {
-            String finalConventionKey = conventionKey;
-            logger.log(() -> "'" + finalConventionKey + "' (original name: '" + key + "') is not a valid tag key. "
+
+        if (conventionKey.startsWith("aws_") || conventionKey.startsWith("gcp_")
+                || conventionKey.startsWith("azure_")) {
+            logger.log("'" + conventionKey + "' (original name: '" + key + "') is not a valid tag key. "
```

Comment:
This may be slower and less performant than the previous implementation when logging is not available.

Reply:
Thanks for catching the change, I had changed this to the existing behaviour.